src/serac/numerics/assembled_sparse_matrix.cpp

```diff
@@ -0,0 +1,263 @@
+
+#include "mfem/linalg/dtensor.hpp"
+#include "assembled_sparse_matrix.hpp"
+
+namespace detail {
+/**
+ * We'd like to access the protected offset, indices, and gatherMap variables from mfem::ElementRestriction
+ * to compute our sparsity patterns in a manner consistent with MFEM.
+ * In order to do that, we use the following template tricks.
+ * Here's an article that explains the trick being used: https://accu.org/journals/overload/28/156/harrison_2776/
+ */
+template <auto offsetV, auto indicesV, auto gatherV>
```

Comment:
Interesting, I missed the `auto` templates when I skimmed this earlier. I thought this was encoding type information into `forbidden_restriction` initially, but it's actually tracking member function pointers. What a weird trick!